File: 백준/Bronze/10798. 세로읽기/세로읽기.java
import java.util.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        List<String> words = new ArrayList<>();
        int maxLen = 0;
        for(int i = 0; i < 5; i++) {
            String word = br.readLine().trim();
            words.add(word);

            if(word.length() > maxLen) {
                maxLen = word.length();
            }
        }

        String result = "";
        for(int i = 0; i < maxLen; i++) {
            for(int j = 0; j < 5; j++) {
                if(i < words.get(j).length()) {
                    result += words.get(j).charAt(i);
                }
            }
        }

        System.out.println(result);
    }
}

File: 백준/Bronze/10798. 세로읽기/세로읽기.py
import sys
input = sys.stdin.readline

words = []

for _ in range(5):
    tmp = input().strip()
    words.append(tmp)

max_len = max(len(word) for word in words)

result = ""
for i in range(max_len):
    for j in range(5):
        if i < len(words[j]):
            result += words[j][i]

print(result)